src/cljs/rems/status_modal.cljs

```diff
@@ -47,6 +47,55 @@
        (when @state [component @state])
        [:button.btn.btn-secondary {:on-click #(reset! state (assoc opened-state :on-close on-close))} "Open modal"]])))
 
+(defn status-modal-opts
+  "Returns a map of modal options, that can be used by status-modal/situational-status-modal.
+
+   The returned value is a map containing various event handlers, and most importantly,
+   a state atom under the key :state-atom."
+  [{:keys [on-success on-pending on-error on-close-after-success on-close-after-error] :as modal-options}]
+  (let [state (r/atom nil)]
+    (merge
+     modal-options
+     {:state-atom state
+      :on-pending #(do
+                     (swap! state assoc :status :pending)
+                     (when on-pending (on-pending)))
+      :on-success #(do
+                     (swap! state assoc :status :saved)
+                     (when on-success (on-success)))
+      :on-error #(do
+                   (swap! state assoc :status :failed :error %)
+                   (when on-error (on-error)))
+      :on-close-after-success #(do
+                                 (reset! state nil)
+                                 (when on-close-after-success (on-close-after-success)))
+      :on-close-after-error #(do
+                               (reset! state nil)
+                               (when on-close-after-error (on-close-after-error)))})))
+
+(defn situational-status-modal
+  "Modal component, that handles closing differently depending on :status
+
+  State is a map containing key :status. If :status is either :saved or :failed,
+  a status-modal is rendered, using either :on-close-after-success,
+  or :on-close-after-error as the :on-close handler."
+  [state {:keys [on-close-after-success on-close-after-error] :as modal-options}]
+  (cond
+    (= :saved (:status state))
+    [status-modal (merge
```

Comment:
This display component can be renamed as well if it's not meant to be used outside?

Reply:
I'm not really sold on the idea that it shouldn't be used outside.

There's 30 cases of usage around rems, and it looks like many of them handle the state of the modal differently. Having them all work the same way would be nice, but I'm not sure if it's worth all the refactoring..

I'd rather just have different entrypoints into the status-modal namespace. That is why I had the "situational-status-modal" component. I agree it's probably not the best name :)

Reply:
There should be one and only one way of showing the status. What is shown is determined by the caller. And there should be only roughly one entrypoint to the status-modal namespace instead of many.